packages/bridge/src/type-templates.ts

```diff
@@ -52,31 +55,98 @@ export const middlewareTypeTemplate = {
   filename: 'types/schema.d.ts',
-  getContents: async ({ nuxt }: TemplateContext) => {
+  getContents: async ({ nuxt }) => {
     const moduleInfo = nuxt.options._installedModules.map(m => ({
       ...m.meta || {},
       importName: m.entryPath || m.meta?.name
     })).filter(m => m.configKey && m.name && !adHocModules.includes(m.name))
 
+    const relativeRoot = relative(resolve(nuxt.options.buildDir, 'types'), nuxt.options.rootDir)
+    const getImportName = (name: string) => (name.startsWith('.') ? './' + join(relativeRoot, name) : name).replace(/\.\w+$/, '')
+    const modules = moduleInfo.map(meta => [genString(meta.configKey), getImportName(meta.importName)])
+
+    // @ts-ignore
+    const nitroEnabled = nuxt.options.bridge?.nitro !== false
+    // For nitro-less build we mirror the runtime config generation to the schema
+    // https://github.com/nuxt/nuxt/blob/5eb1b32f62a0ad92bfa6f37641489c35caa4b791/packages/vue-renderer/src/renderer.js#L300
+
+    // @ts-ignore
+    const runtimeConfigs = nitroEnabled
+      ? {
+          private: Object.fromEntries(Object.entries(nuxt.options.runtimeConfig).filter(([key]) => key !== 'public')),
+          public: nuxt.options.runtimeConfig.public
+        }
+
+      : {
+          // @ts-ignore
+          private: defu(nuxt.options.privateRuntimeConfig, nuxt.options.publicRuntimeConfig),
+          // @ts-ignore
+          public: nuxt.options.publicRuntimeConfig
+        }
+
+    const generatedPrivateTypes = generateTypes(await resolveSchema(runtimeConfigs.private as Record<string, JSValue>),
+      {
+        interfaceName: 'RuntimeConfig',
+        addExport: false,
+        addDefaults: false,
+        allowExtraKeys: false,
+        indentation: 2
+      })
+    const generatedPublicTypes = generateTypes(await resolveSchema(runtimeConfigs.public as Record<string, JSValue>),
+      {
+        interfaceName: 'PublicRuntimeConfig',
+        addExport: false,
+        addDefaults: false,
+        allowExtraKeys: false,
+        indentation: 2
+      })
+
+    const vueTypesConfig = nitroEnabled
+      ? ['declare module \'vue/types/vue\' {',
+          generatedPrivateTypes,
+          '  interface Vue {',
+          '    $config: Omit<RuntimeConfig & { public: PublicRuntimeConfig }, \'_app\'>',
+          '  }',
+          '}'
+        ].join('\n')
+      : [
+          'declare module \'vue/types/vue\' {',
+          generatedPrivateTypes,
+          generatedPublicTypes,
+          '  interface Vue {',
+          '    $config: Omit<RuntimeConfig, \'public\'>',
+          '  }',
+          '}'
+        ].join('\n')
+
+    const bridgeSchemaConfig = [
+      'declare module \'@nuxt/bridge-schema\' {',
+      '  interface NuxtApp {',
+      '    $config: RuntimeConfig',
+      '  }',
+      '  interface NuxtAppCompat {',
+      '    $config: RuntimeConfig',
+      '  }',
+      '  interface NuxtRuntimeConfig extends RuntimeConfig {}',
```

Comment:
I don't really know about this: we can use 
```ts
const { $config } = useNuxtApp()

const { nuxt2Context: { $config } } = useNuxtApp()
```

but displayed types will be different